employee-rostering-frontend/cypress/integration/newTenantBasicWorkflow.js

```diff
@@ -0,0 +1,166 @@
+       cy.get('[data-cy=settings]').click();
+       cy.get('[data-cy=reset-application]').click();
+       closeAlerts();
+    });
+    
+    it('basic tenant workflow', () => {
+        // Create a tenant
+        cy.get('[data-cy=settings]').click();
+        cy.get('[data-cy=add-tenant]').click();
+        cy.get('[data-cy=name]').type("Test Tenant");
+        cy.get('[data-cy=schedule-start-date]').type("2018-01-01");
+        cy.get('[data-cy=draft-length]').type("7");
+        cy.get('[data-cy=publish-notice]').type("7");
+        // publish length is disabled/not supported by backend
+        // cy.get('[data-cy=publish-length]').type("7");
+        cy.get('[data-cy=rotation-length]').type("7");
+        selectValue("Select a timezone...", "UTC");
+        cy.get('[data-cy=save]').click();
+        closeAlerts();
+        
+        changeToTenant("Test Tenant");
+        
+        // Create a new skill
+        gotoPage("skills");
+        cy.get("button").contains("Add").click();
+        cy.get('[aria-label="Name"]').type("New Skill");
+        cy.get('[aria-label="Save"]').click();
+        closeAlerts();
+        
+        // Create two spots
+        gotoPage("spots");
+        
+        // First spot doesn't require any skills
+        cy.get("button").contains("Add").click();
+        cy.get('[aria-label="Name"]').type("No Skill Spot");
+        cy.get('[aria-label="Save"]').click();
+        closeAlerts();
+        
+        // Second spot requires our skill
+        cy.get("button").contains("Add").click();
+        cy.get('[aria-label="Name"]').type("Required Skill Spot");
+        selectValue("Select required skills...", "New Skill");
+        cy.get('[aria-label="Save"]').click();
+        closeAlerts();
+        
+        // Create a contract
+        gotoPage("contracts");
+        cy.get("button").contains("Add").click();
+        cy.get('[aria-label="Name"]').type("New Contract");
+        cy.get('[aria-label="Save"]').click();
+        closeAlerts();
+        
+        // Create two employees
+        gotoPage("employees");
+        
+        // First employee doesn't have any skills
+        cy.get("button").contains("Add").click();
+        cy.get('[aria-label="Name"]').type("No Skills Employee");
+        selectValue("Select a contract...", "New Contract");
+        cy.get('[aria-label="Save"]').click();
+        
+        // Second employee has our skill
+        cy.get("button").contains("Add").click();
+        cy.get('[aria-label="Name"]').type("Employee with Skills");
+        selectValue("Select a contract...", "New Contract");
+        selectValue("Select skill proficiencies...", "New Skill");
+        cy.get('[aria-label="Save"]').click();
+        closeAlerts();
+        
+        // Create some shifts
+        gotoPage("shift");
+        
+        // First shift is on Monday, and is for the spot that doesn't require any skills
+        selectValue("Select a Spot...", "No Skill Spot");
+        cy.wait(1000);
```

Comment:
I wonder what is exactly happening here that requires waiting. I was unable to get rid of this 1 sec.

Reply:
Waiting for the frontend to re-render (the roster is a giant JSON, so it take time)

Reply:
Would it be able to assert value of some element that is guaranteed to show up once the re-rendering is finished? This way Cypress would periodically check presence of the element with a timeout, which is in my opinion more efficient and reliable than waiting for a fixed amount of time.

Reply:
It now waits for the value of the input to be updated